docs/root/intro/life_of_a_request.rst

```diff
@@ -0,0 +1,631 @@
+   capacity.
+8. The upstream endpoint connection's HTTP/2 codec multiplexes the request’s stream with any other
+   streams going to that upstream over a single TCP connection.
+9. The upstream endpoint connection's TLS transport socket encrypts these bytes and writes them to a
+   TCP socket for the upstream connection.
+10. The request, consisting of headers, and optional body and trailers, is proxied upstream, and the
+    response is proxied downstream. The response passes through the HTTP filters in the opposite
+    order from the request, starting at the router filter and passing through CustomFilter, before
+    being sent downstream.
+11. When the response is complete, the stream is destroyed. Post-request processing will update
+    stats, write to the access log and finalize trace spans.
+
+We elaborate on each of these steps in the sections below.
+
+1. Listener TCP accept
+^^^^^^^^^^^^^^^^^^^^^^
+
+.. image:: /_static/lor-listeners.svg
+   :width: 90%
+   :align: center
+
+The *ListenerManager* is responsible for taking configuration representing :ref:`listeners
+<arch_overview_listeners>` and instantiating a number of *Listener* instances bound to their
+respective IP/ports. Listeners may be in one of three states:
+
+* *Warming*: the listener is waiting for configuration dependencies (e.g. route configuration,
+  dynamic secrets). The listener is not yet ready to accept TCP connections.
+* *Active*: the listener is bound to its IP/port and accepts TCP connections.
+* *Draining*: the listener no longer accepts new TCP connections while its existing TCP connections
+  are allowed to continue for a drain period.
+
+Each :ref:`worker thread <arch_overview_threading>` maintains its own *Listener* instance for each
+of the configured listeners. Each listener may bind to the same port via SO_REUSEPORT or share a
+single socket bound to this port. When a new TCP connection arrives, the kernel decides which
+worker thread will accept the connection and the *Listener* for this worker thread will have its
+``Server::ConnectionHandlerImpl::ActiveTcpListener::onAccept()`` callback invoked.
+
+2. Listener filter chains and network filter chain matching
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The worker thread’s *Listener* then creates and runs the :ref:`listener filter
+<arch_overview_listener_filters>` chain. Filter chains are created by applying each filter’s *filter
+factory*. The factory is aware of the filter’s configuration and creates a new instance of the
+filter for each connection or stream.
+
+In the case of our TLS listener configuration, the listener filter chain consists of the :ref:`TLS
+inspector <config_listener_filters_tls_inspector>` filter
+(``envoy.filters.listener.tls_inspector``). This filter examines the initial TLS handshake and
+extracts the server name (SNI). The SNI is then made available for filter chain matching. The TLS
```

Comment:
I think we should put it explicitly in the configuration in this example. I think it leads to better understanding of what is going on. And if you need other listener filters (PROXY protocol for instance), they should always be configured explicitly to ensure they're in the correct order.